types/module/configurator.go

```diff
@@ -38,3 +55,54 @@ func (c configurator) MsgServer() grpc.Server {
 func (c configurator) QueryServer() grpc.Server {
 	return c.queryServer
 }
+
+// RegisterMigration implements the Configurator.RegisterMigration method
+func (c configurator) RegisterMigration(moduleName string, forVersion uint64, handler MigrationHandler) error {
+	if forVersion == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidVersion, "module migration versions should start at 1")
+	}
+
+	if c.migrations[moduleName] == nil {
+		c.migrations[moduleName] = map[uint64]MigrationHandler{}
+	}
+
+	if c.migrations[moduleName][forVersion] != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrLogic, "another migration for module %s and version %d already exists", moduleName, forVersion)
+	}
+
+	c.migrations[moduleName][forVersion] = handler
+
+	return nil
+}
+
+// runModuleMigrations runs all in-place store migrations for one given module from a
+// version to another version.
+func (c configurator) runModuleMigrations(ctx sdk.Context, moduleName string, fromVersion, toVersion uint64) error {
+	// No-op if toVersion is the initial version.
+	// Some modules don't have a store key (e.g. vesting), in this case, their
+	// ConsensusVersion will always stay at 1, and running migrations on
```

Comment:
I don't think this is strictly true. Vesting doesn't have its own storage (currently), but its state machine _may_ change. Do we need to distinguish "state version" from "state machine version"? The idea I had was that `ConsensusVersion` represents any change to the state store and/or state machine. In some cases the `ConsensusVersion` may increment but there are no migrations so a no-op migration should be registered. Does that make sense?

Reply:
Got it, yeah, I initially went with ConsensusVersion bump == needs store migration. Your idea makes sense, I updated the code.